src/audio/drc/drc_math_generic.c

```diff
@@ -105,16 +123,170 @@ inline int32_t drc_log_fixed(int32_t x)
 }
 
 /*
- * Input is Q2.30: (-2.0, 2.0)
- * Output range: (-1.0, 1.0); regulated to Q1.31: (-1.0, 1.0)
+ * function [cdcSinTh] = drc_sin_fixed(thRadFxp)
+ * compute the sine theta.thRadFxp contains angle values in radians in the range of [-2pi,2pi]
+ * cdcSinTh is the CORDIC-based approximation of the sine of theta
+ * rotate the vector over given angle. vector co ordinate ( x & y ) is rotated through angle
+ * and obtain new vector co ordinate.
+ * Use Q12.20-bit quantized inputs and with max error 0.
+ *	 Compare the fixed-point cordicsin function results to the
+ *	 results of the double-precision sin function.
+ *
+ * +--------------+---------+----------+----------+---------+
+ * |					|thRadFxp |cdcSinTh	|QthRadFxp |QcdcSinTh|
+ * +--------------+---------+----------+----------+---------+
+ * |	  WordLength| 31		 |	 32		|12.20	  | 3.29		|
+ * |FractionLength| 20		 |	 29		|			  |			|
+ * +--------------+---------+----------+----------+---------+
+ *	 THD+N = 29*6 = -174 dBc
+ * Arguments	 : int32_t thRadFxp Q12.20 input
+ *						int32_t cdcSinTh Q3.29	output
+ * Return Type	 : cdcSinTh
  */
-inline int32_t drc_sin_fixed(int32_t x)
+inline int32_t drc_sin_fixed(int32_t thRadFxp)
 {
-	const int32_t PI_OVER_TWO = Q_CONVERT_FLOAT(1.57079632679489661923f, 30);
-
-	/* input range of sin_fixed() is non-negative */
-	int32_t abs_sin_val = sin_fixed(q_mult(ABS(x), PI_OVER_TWO, 30, 30, 28));
-	return SGN(x) < 0 ? -abs_sin_val : abs_sin_val;
+	/* cordic_atan2_lookup_table = atan(2.^-(0:N-1)) N = 30	*/
+	/* CORDIC Gain is cordic_gain = prod(sqrt(1 + 2.^(-2*(0:30-1)))) */
+	/* Inverse CORDIC Gain,inverse_cordic_gain = 1 / cordic_gain  */
+  static const int32_t inpLUT[30] = { 421657428, 248918915, 131521918, 66762579,
+	 33510843, 16771758, 8387925, 4194219, 2097141, 1048575, 524288, 262144,
+	 131072, 65536, 32768, 16384, 8192, 4096, 2048, 1024, 512, 256, 128, 64, 32,
+	 16, 8, 4, 2, 1 };
+
+  int32_t b_idx;
+  int32_t b_yn;
+  int32_t th;
+  int32_t xn;
+  int32_t xtmp;
+  int32_t ytmp;
+  int32_t z;
+  int32_t cdcSinTh;
+  bool negate;
+
+  th = thRadFxp;
+	 if (th > PIQ20OFFSET0) {											 /* >20bits */
```

Comment:
The indentation looks off here, should be tab size 8 (but sometimes GH does not show this correctly)

Reply:
I find if github is off usually that is a sign of tabs v spaces

Reply:
checkpatch reports many indentation issues.